lib/templates/_footer.html

```diff
@@ -56,10 +56,31 @@
   scrollBy(0, -68);
 }
 
+function initSourceToggle() {
+  var toggle = document.querySelector('.source-code h4');
+  var toggleArrow = document.querySelector('.source-code-toggle');
+  var code = document.querySelector('.source-code pre');
+
+  if (toggle && toggleArrow && code) {
+    toggle.addEventListener('click', function(e) {
+      if (toggleArrow.classList.contains('closed')) {
```

Comment:
This might be a bit more terse:
```
bool closed = toggleArrow.classes.contains('closed');
toggleArrow.classes.toggle('closed', !closed);
toggleArrow.classes.toggle('open', closed);
code.style.display = closed ? 'block' : 'none';
```